services/motion/builtin/move_request.go

```diff
@@ -124,16 +125,27 @@ func (mr *moveRequest) execute(ctx context.Context, waypoints state.Waypoints, w
 	for i := int(waypointIndex.Load()); i < len(waypoints); i++ {
 		select {
 		case <-ctx.Done():
+			mr.logger.Debugf("calling kinematicBase.Stop due to %s\n", ctx.Err())
+			stopCtx, cancelFn := context.WithTimeout(context.Background(), baseStopTimeout)
+			if stopErr := mr.kinematicBase.Stop(stopCtx, nil); stopErr != nil {
```

Comment:
I think we can defer the cancelFn here instead

Reply:
the linter won't let me, I'll paste the error in a moment

Reply:
done